cdap-docs/developers-manual/source/building-blocks/schedules.rst

```diff
@@ -23,8 +34,96 @@ Optionally, you can specify the properties for the Schedule::
+                     "MyWorkflow");
+
+The ``purchaseStream`` will either have to exist in CDAP when deploying your application, or you will have to
+`add it <../../reference-manual/javadocs/co/cask/cdap/api/app/AbstractApplication.html#addStream(co.cask.cdap.api.data.stream.Stream)>`__
+when configuring your application. If you try to create a **Stream Size Schedule** on a Stream which does not exist,
+your application will fail to deploy.
+
+Optionally, you can specify the properties for the Schedule::
+
+    ...
+    Map<String, String> scheduleProperties = Maps.newHashMap();
+    scheduleProperties.put("myProperty", "10");
+    scheduleProperties.put("anotherProperty", "anotherValue");
+
+    scheduleWorkflow(Schedules.createDataSchedule("1MBStreamSchedule", "Schedule triggered every 1MB of ingested data",
+                                                  Schedules.Source.STREAM, "purchaseStream", 1),
+                     "MyWorkflow", scheduleProperties);
+    ...
+
+Every time the ``1MBStreamSchedule`` triggers, it passes the ``scheduleProperties`` as runtime arguments to the ``MyWorkflow``.
+The properties specified by the Schedule can be overridden by the Workflow's runtime arguments, with the Workflow's having precedence.
+
+The **Stream Size Schedule** uses two sources of information to know how much data has been ingested by a Stream.
+One is :ref:`Stream size notifications <streams-notifications>`, and the other one is the aggregated metric that
+records the total size of events ingested by a Stream.
+Notifications are only used as an indication that data is available. Stream Size Schedules always confirm there
+information by querying the Metric system, which is the reliable source of information.
+
+
+.. rubric:: Stream Size Schedules Lifecycle
+
+A **Stream Size Schedule** will execute a Workflow every time the Stream it references ingests an increment of data,
+also defined in the
+`Schedule. <../../reference-manual/javadocs/co/cask/cdap/api/schedule/Schedules.html#createDataSchedule(java.lang.String,%20java.lang.String,%20co.cask.cdap.api.schedule.Source,%20java.lang.String,%20int)>`__
+
+A few actions can be performed on a Schedule:
+
+- *Suspend*: suspending a schedule will prevent the Workflow from being executed again, in the event that the stream
+  ingests enough data. This action will not suspend the current execution of the Workflow, if the Schedule triggered
+  it recently. Suspending an already suspended Schedule has no effect;
+- *Resume*: this action will put the Schedule back into an active state, after a *Suspend* action. Resuming an already
+  active Schedule has no effect. If the Stream has ingested more than the increment of data defined by the Schedule
+  since the last time the Schedule was triggered, while the Schedule was suspended, then the Workflow will be executed
+  upon resume;
+- *Update*: this action is triggered when an application which contains a Stream Size Schedule is redeployed in CDAP.
```

Comment:
*Resume*: This action will put the Schedule back into an active state, after a *Suspend* action. Resuming an already-active Schedule has no effect. If—while the Schedule was suspended, and since the last time the Schedule was triggered—the Stream has ingested more than the increment of data defined by the Schedule, the Workflow will be 
immediately executed upon resume.